metadata-ingestion/src/datahub/ingestion/transformer/pattern_cleanup_ownership.py

```diff
@@ -58,11 +70,16 @@ def transform_aspect(
         # clean all the owners based on the parameters received from config
         cleaned_owner_urns: List[str] = []
         for owner_urn in current_owner_urns:
-            user_id: str = owner_urn.split(_USER_URN_PREFIX)[1]
-            for value in self.config.pattern_for_cleanup:
-                user_id = re.sub(value, "", user_id)
-
-            cleaned_owner_urns.append(_USER_URN_PREFIX + user_id)
+            try:
+                owner: Urn = Urn.from_string(owner_urn)
+                if isinstance(owner, CorpUserUrn):
+                    cleaned_owner_urns.append(self._process_owner(owner.username))
+                elif isinstance(owner, CorpGroupUrn):
+                    cleaned_owner_urns.append(self._process_owner(owner.name))
```

Comment:
does this convert group urns to user urns?

Reply:
actually it just converts all urns to invalid urns! That's big brain coding right there.